File: Algorithms_medium/6_Zigzag Conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 0:
            return ""
        if numRows == 1:
            return s
        List  = [ [] for i in range(numRows) ]
        i = 0
        while i < len(s):
            j = 0
            while i < len(s) and j < numRows:
                List[j].append(s[i])
                i += 1
                j += 1
            j -= 2
            while i < len(s) and j > 0:
                List[j].append(s[i])
                j -= 1
                i += 1
                
        return "".join(["".join(n) for n in List])
'''
按照题目要求打出一个数字表，然后返还数列数字，用join消除中间的空格
'''

File: Algorithms_C#/6_Zigzag_Conversion.cs
public class Solution {
    public string Convert(string s, int numRows) {
        int n = s.Length, r = numRows;
        if (r == 1 || r >= n){
            return s;
        }
        String ans = null;
        int t = 2 * r - 2;
        for (int i = 0; i < r; ++i){
            for (int j = 0; j+i < n; j += t){
                ans += s[j + i];
                if (0 < i && i < r - 1 && j + t - i < n){
                    ans += s[j + t - i];
                }
            }
        }
        return ans;
    }
}